### ISSUE TITLE
Cosmic Store crash on launch

### ISSUE THREAD

**Issue by aBgrOnGak** [pop-os/cosmic-store#71]
Hello and good day.

As per title, it did not give any warning whatsoever.

How do I inspect any logs or something? FYI, I removed pop shop before installing cosmic store.

**Comment by mehedi-iut** [comment]
I am also facing the same issue
output of the of **cosmic-store** command
```
2024-05-14T04:41:59Z ERROR i18n_embed::fluent] Error while adding resource to bundle: Overriding { kind: Message, id: "utilities" }.
[2024-05-14T04:41:59Z ERROR i18n_embed::fluent] Error while adding resource to bundle: Overriding { kind: Message, id: "utilities" }.
[2024-05-14T04:41:59Z WARN  wgpu_hal::gles::egl] No config found!
[2024-05-14T04:41:59Z WARN  wgpu_hal::gles::egl] EGL says it can present to the window but not natively
[2024-05-14T04:41:59Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadMatch (invalid parameter attributes)",
        error_code: 8,
        request_code: 149,
        minor_code: 4,
    }
[2024-05-14T04:41:59Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadMatch (invalid parameter attributes)",
        error_code: 8,
        request_code: 149,
        minor_code: 4,
    }
[2024-05-14T04:41:59Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadMatch (invalid parameter attributes)",
        error_code: 8,
        request_code: 149,
        minor_code: 4,
    }
thread 'main' panicked at /build/cosmic-store-FpsAEw/cosmic-store-0.1.0~1714166022~22.04~225ebca/vendor/winit/src/platform_impl/linux/x11/window.rs:1208:18:
Failed to call XMapRaised: XError { description: "BadMatch (invalid parameter attributes)", error_code: 8, request_code: 149, minor_code: 4 }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /build/cosmic-store-FpsAEw/cosmic-store-0.1.0~1714166022~22.04~225ebca/vendor/winit/src/platform_impl/linux/x11/window.rs:601:34:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
stack backtrace:
   0:     0x59a55684315b - <unknown>
   1:     0x59a5568899e0 - <unknown>
   2:     0x59a556836abd - <unknown>
   3:     0x59a556842f44 - <unknown>
   4:     0x59a55686c21a - <unknown>
   5:     0x59a55686bebd - <unknown>
   6:     0x59a55686c638 - <unknown>
   7:     0x59a55684353e - <unknown>
   8:     0x59a556843376 - <unknown>
   9:     0x59a55686c362 - <unknown>
  10:     0x59a55577a665 - <unknown>
  11:     0x59a55577ae13 - <unknown>
  12:     0x59a555f32203 - <unknown>
  13:     0x59a555f0fa6c - <unknown>
  14:     0x59a5559835f4 - <unknown>
  15:     0x59a5558e902c - <unknown>
  16:     0x59a5558f4bc4 - <unknown>
  17:     0x59a5559b0899 - <unknown>
  18:     0x59a5558ef865 - <unknown>
  19:     0x59a55583628e - <unknown>
  20:     0x59a5559f4013 - <unknown>
  21:     0x59a5558f2ba9 - <unknown>
  22:     0x59a55686c254 - <unknown>
  23:     0x59a5568522eb - <unknown>
  24:     0x59a5558f2b9e - <unknown>
  25:     0x7d4999c29d90 - <unknown>
  26:     0x7d4999c29e40 - __libc_start_main
  27:     0x59a55577b245 - <unknown>
  28:                0x0 - <unknown>
thread 'main' panicked at library/core/src/panicking.rs:144:5:
panic in a destructor during cleanup
thread caused non-unwinding panic. aborting.
[1]    31605 IOT instruction (core dumped)  cosmic-store

```

**Comment by jjinco33** [comment]
Impacted by same, did not try switching to Wayland. If needed/matters:

$ cosmic-store 
[2024-05-12T03:55:36Z ERROR i18n_embed::fluent] Error while adding resource to bundle: Overriding { kind: Message, id: "utilities" }.
[2024-05-12T03:55:36Z ERROR i18n_embed::fluent] Error while adding resource to bundle: Overriding { kind: Message, id: "utilities" }.
[2024-05-12T03:55:36Z WARN  wgpu_hal::gles::egl] No windowing system present. Using surfaceless platform
[2024-05-12T03:55:36Z WARN  wgpu_hal::gles::egl] No config found!
[2024-05-12T03:55:36Z WARN  wgpu_hal::gles::egl] No config found!
[2024-05-12T03:55:36Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadMatch (invalid parameter attributes)",
        error_code: 8,
        request_code: 149,
        minor_code: 4,
    }
[2024-05-12T03:55:36Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadMatch (invalid parameter attributes)",
        error_code: 8,
        request_code: 149,
        minor_code: 4,
    }
[2024-05-12T03:55:36Z ERROR winit::platform_impl::platform] X11 error: XError {
        description: "BadMatch (invalid parameter attributes)",
        error_code: 8,
        request_code: 149,
        minor_code: 4,
    }
thread 'main' panicked at /build/cosmic-store-FpsAEw/cosmic-store-0.1.0~1714166022~22.04~225ebca/vendor/winit/src/platform_impl/linux/x11/window.rs:1208:18:
Failed to call XMapRaised: XError { description: "BadMatch (invalid parameter attributes)", error_code: 8, request_code: 149, minor_code: 4 }
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
thread 'main' panicked at /build/cosmic-store-FpsAEw/cosmic-store-0.1.0~1714166022~22.04~225ebca/vendor/winit/src/platform_impl/linux/x11/window.rs:601:34:
called `Result::unwrap()` on an `Err` value: PoisonError { .. }
stack backtrace:
   0:     0x55c43081715b - <unknown>
   1:     0x55c43085d9e0 - <unknown>
   2:     0x55c43080aabd - <unknown>
   3:     0x55c430816f44 - <unknown>
   4:     0x55c43084021a - <unknown>
   5:     0x55c43083febd - <unknown>
   6:     0x55c430840638 - <unknown>
   7:     0x55c43081753e - <unknown>
   8:     0x55c430817376 - <unknown>
   9:     0x55c430840362 - <unknown>
  10:     0x55c42f74e665 - <unknown>
  11:     0x55c42f74ee13 - <unknown>
  12:     0x55c42ff06203 - <unknown>
  13:     0x55c42fee3a6c - <unknown>
  14:     0x55c42f9575f4 - <unknown>
  15:     0x55c42f8bd02c - <unknown>
  16:     0x55c42f8c8bc4 - <unknown>
  17:     0x55c42f984899 - <unknown>
  18:     0x55c42f8c3865 - <unknown>
  19:     0x55c42f80a28e - <unknown>
  20:     0x55c42f9c8013 - <unknown>
  21:     0x55c42f8c6ba9 - <unknown>
  22:     0x55c430840254 - <unknown>
  23:     0x55c4308262eb - <unknown>
  24:     0x55c42f8c6b9e - <unknown>
  25:     0x76092a029d90 - __libc_start_call_main
                               at ./csu/../sysdeps/nptl/libc_start_call_main.h:58:16
  26:     0x76092a029e40 - __libc_start_main_impl
                               at ./csu/../csu/libc-start.c:392:3
  27:     0x55c42f74f245 - <unknown>
  28:                0x0 - <unknown>
thread 'main' panicked at library/core/src/panicking.rs:144:5:
panic in a destructor during cleanup
thread caused non-unwinding panic. aborting.
Aborted (core dumped)

**Comment by jjinco33** [comment]
Issue appears resolved now.

**Comment by papercutter0324** [comment]
I just installed Pop_OS 22.04 a few days ago and came across this problem. Under Wayland, Cosmic-Store opens no problem; but under x11, it absolutely refused to. In my search for a soloution, I came across a Reddit post by a user with the exact same problem. I will include their error log (convenient since I still have that page open), but it is near identical to mine.

I didn't run the back trace, but one of the comments had the exact solution I needed. By running the command 'env WGPU_ADAPTER_NAME=3070 cosmic-store', it loaded up under X11 no problem.

Error log from Reddit:

[2024-06-21T14:50:18Z WARN wgpu_hal::gles::egl] No windowing system present. Using surfaceless platform

[2024-06-21T14:50:18Z WARN wgpu_hal::gles::egl] No config found!

[2024-06-21T14:50:18Z WARN wgpu_hal::gles::egl] No config found!

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "BadMatch (invalid parameter attributes)",

error_code: 8,

request_code: 149,

minor_code: 4,

}

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "BadMatch (invalid parameter attributes)",

error_code: 8,

request_code: 149,

minor_code: 4,

}

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "BadMatch (invalid parameter attributes)",

error_code: 8,

request_code: 149,

minor_code: 4,

}

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "BadMatch (invalid parameter attributes)",

error_code: 8,

request_code: 149,

minor_code: 4,

}

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "BadMatch (invalid parameter attributes)",

error_code: 8,

request_code: 149,

minor_code: 4,

}

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "BadMatch (invalid parameter attributes)",

error_code: 8,

request_code: 149,

minor_code: 4,

}

[2024-06-21T14:50:18Z ERROR winit::platform_impl::platform] X11 error: XError {

description: "136",

error_code: 136,

request_code: 148,

minor_code: 1,

}

[2024-06-21T14:50:19Z ERROR iced_winit::multi_window] Error Timeout when presenting surface.

[2024-06-21T14:50:20Z ERROR iced_winit::multi_window] Error Timeout when presenting surface.

[2024-06-21T14:50:21Z ERROR iced_winit::multi_window] Error Timeout when presenting surface.

[2024-06-21T14:50:22Z ERROR iced_winit::multi_window] Error Timeout when presenting surface.

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

[2024-06-21T14:50:22Z WARN iced_futures::subscription::tracker] Error sending event to subscription: TrySendError { kind: Full }

thread 'main' panicked at /build/cosmic-store-dR5whJ/cosmic-store-0.1.0~1718281780~22.04~ba3452a/vendor/winit/src/platform_impl/linux/x11/event_processor.rs:1042:30:

Failed to unfocus input context: XError { description: "136", error_code: 136, request_code: 148, minor_code: 1 }

note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
